### PR TITLE
[READY?!?!?] Fragmentation overhaul

### PR DESCRIPTION
**Pull request by kevingranade**
What a haul, this took many times longer than expected, but it's ready for checking out, I've exhausted the issues I know about.

Main features:
1. Fragmentation lethality based on real-world data, centered on standard antipersonnel munitions.
  Specifically, added tests to verify that M87 is highly lethal at 5m, and drops sharply in lethality at 15m.
2. Explosives stats and effectiveness derived from real-world data. (Casing weight and explosive charge mass)
3. Fragment cloud algorithm based on shadowcasting, providing a fast but accurate way to model a cloud made up of huge numbers of fragments.
  Interacts with obstacles, including monsters, leading to fragmentation effectiveness scaling nicely based on openness of terrain.
4. Goosed mininuke power and stats to resemble smallest known operational nuke.
 
Some discussion of the changes on the [forums](https://discourse.cataclysmdda.org/t/best-wmds-for-killing-large-groups-of-z/15623?u=kevin.granade).

Some RDX charge tests:
https://discourse-cdn-sjc2.com/standard15/uploads/cataclysmdda/original/2X/3/381595cec2e35e0985f9204c500bad19806ea9b3.png
https://discourse-cdn-sjc2.com/standard15/uploads/cataclysmdda/original/2X/1/171351ccca9b12d8136c2dca503232a504a2a916.png

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -7501,6 +7500,74 @@ void map::build_outside_cache( const int zlev )
...
+        for( size_t part = 0; part < v.v->parts.size(); part++ ) {
+            int px = v.x + v.v->parts[part].precalc[0].x;
+            int py = v.y + v.v->parts[part].precalc[0].y;
+            int pz = v.z + OVERMAP_DEPTH;
+            // TODO: Replace this with a start/end bounds check.
+            if( !inbounds( px, py ) ) {
+                continue;
+            }
+
+            if( v.v->obstacle_at_part( part ) >= 0 ) {
```
Comment (ZhilkinSerg) [inline src/map.cpp]: Will need changes due to #23873.

**Comment by ZhilkinSerg** [comment]
Lol, I've just blasted a prison, a hospital and a football field with a mininuke and that was something!

**Comment by nexusmrsep** [comment]
That might be an incentive for a new set of home made explosive traps/mines.

**Comment by kevingranade** [comment]
Need to consider ranges.
1. Verify that fuses burned for long enough for the player to get safely away.
2. Verify that an average Survivor can throw Thrown explosives far enough to exceed the minimum safe distance.

**Comment by Clyybber** [comment]
Probably for a different PR, but fuses which the player is able to lay on the ground would be cool, allowing a remote detonation.

**Comment by kevingranade** [comment]
> Couldn't heavy duty cable code be used for this?

This is actually a lot simpler than what vehicle power cables do, so I'd prefer to not tie those two together.

Opened another issue for fuses: #24556

**Comment by kevingranade** [comment]
Tweaked throw distances (but not accuracy) to make grenades usable without killing yourself.
Still need to adjust fuze lengths.

**Comment by kevingranade** [comment]
Fuse times of large charges adjusted so you have time to get clear.

Should be good to go unless something bad crops up.

**Pull request merged**